includes/Block/Web_Stories_Block.php

```diff
@@ -0,0 +1,409 @@
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Block;
+
+use Google\Web_Stories\Embed_Base;
+use Google\Web_Stories\Story_Query;
+use Google\Web_Stories\Tracking;
+use Google\Web_Stories\Story_Post_Type;
+use Google\Web_Stories\Traits\Assets;
+use WP_Post_Type;
+
+/**
+ * Latest Stories block class.
+ */
+class Web_Stories_Block {
+	use Assets;
+
+	/**
+	 * Script handle.
+	 *
+	 * @var string
+	 */
+	const SCRIPT_HANDLE = 'web-stories-block';
+
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	const BLOCK_NAME = 'web-stories/list';
+
+	/**
+	 * Current block's block attributes.
+	 *
+	 * @var array Block Attributes.
+	 */
+	protected $block_attributes = [];
+
+	/**
+	 * Maximum number of stories users can select
+	 *
+	 * @var int
+	 */
+	protected $max_num_of_stories = 20;
+
+	/**
+	 * Initializes the Web Stories embed block.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return void
+	 */
+	public function init() {
+		$this->register_script( self::SCRIPT_HANDLE, [ Embed_Base::STORY_PLAYER_HANDLE, Tracking::SCRIPT_HANDLE ] );
+		$this->register_style( self::SCRIPT_HANDLE );
+
+		wp_localize_script(
+			self::SCRIPT_HANDLE,
+			'webStoriesBlockSettings',
+			$this->get_script_settings()
+		);
+
+		// Note: does not use 'script' and 'style' args, and instead uses 'render_callback'
+		// to enqueue these assets only when needed.
+		register_block_type(
+			self::BLOCK_NAME,
+			[
+				'attributes'      => [
+					'blockType'        => [
+						'type' => 'string',
+					],
+					'url'              => [
+						'type' => 'string',
+					],
+					'title'            => [
+						'type'    => 'string',
+						'default' => __( 'Web Story', 'web-stories' ),
+					],
+					'poster'           => [
+						'type' => 'string',
+					],
+					'width'            => [
+						'type'    => 'number',
+						'default' => 360,
+					],
+					'height'           => [
+						'type'    => 'number',
+						'default' => 600,
+					],
+					'align'            => [
+						'type'    => 'string',
+						'default' => 'none',
+					],
+					'stories'          => [
+						'type'    => 'array',
+						'default' => [],
+					],
+					'viewType'         => [
+						'type'    => 'string',
+						'default' => '',
+					],
+					'numOfStories'     => [
+						'type'    => 'number',
+						'default' => 5,
+					],
+					'numOfColumns'     => [
+						'type'    => 'number',
+						'default' => 2,
+					],
+					'sizeOfCircles'    => [
+						'type'    => 'number',
+						'default' => 96,
+					],
+					'orderByValue'     => [
+						'type'    => 'string',
+						'default' => '',
+					],
+					'viewAllLinkLabel' => [
+						'type'    => 'string',
+						'default' => __( 'View all stories', 'web-stories' ),
+					],
+					'authors'          => [
+						'type'    => 'array',
+						'default' => [],
+					],
+					'fieldState'       => [
+						'type'    => 'object',
+						'default' => $this->fields_states(),
+					],
```

Comment:
Why is `fieldState` a block attribute? Seems unnecessary?

Reply:
This is used to check the initial state for each of the view-types and if user updates any of the available values we can store it in the same object per view. This will keep the user preference intact even if user switches between different views.

Also the same is passed down to the renderer as block values.

Reply:
This initial state - can't it be provided via `wp_localize_script` or something?

Storing this initial state as a block attribute (which is persisted in the database) feels wrong.

Reply:
agree with you there, I'll see to update it so that the block attributes have only the required data and rest will try to use localized data for getting view states.

Reply:
thanks!

Reply:
@swissspidy I've updated the PR to use only required data for the block. 
- Removed default attribute `fieldState` value and instead will be using `localized` values to set the initial state on the component mount.
- Also now the `fieldState` attribute will have values like `show_title`, `show_date` etc instead of the entire state object.